### PR TITLE
fix: load webpack hook before config is required

### LINKED ISSUE #21679: Error: Cannot find module 'webpack'
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

14.15.4

**What browser are you using?**

Safari

**What operating system are you using?**

macOS

**How are you deploying your application?**

yarn build locally

**Describe the Bug**

I just updated to 10.0.6 version from 10.0.5 and im getting the next error: Error: Cannot find module 'webpack'.

I found out that if I remove next-pwa it works, and the terminal output while building all of the errors came form next-pwa.

**Expected Behavior**

If I downgrade to 10.0.5 my project builds in dev and prod.

**To Reproduce**

Install the latest version of next and install next-pwa.

`
yarn install v1.22.10
info No lockfile found.
[1/4] 🔍  Resolving packages...
warning next-pwa > workbox-webpack-plugin > workbox-build > rollup > fsevents@2.1.3: "Please update to latest v2.3 or v2.2"
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi@16.1.8: Switch to 'npm install joi'
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/formula@1.2.0: Moved to 'npm install @sideway/formula'
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/topo > @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
warning next-pwa > workbox-webpack-plugin > workbox-build > @hapi/joi > @hapi/pinpoint@1.0.2: Moved to 'npm install @sideway/pinpoint'
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > next-pwa@5.0.4" has unmet peer dependency "webpack@>=4.0.0".
warning "next-pwa > babel-loader@8.2.2" has unmet peer dependency "@babel/core@^7.0.0".
warning "next-pwa > babel-loader@8.2.2" has unmet peer dependency "webpack@>=2".
warning "next-pwa > workbox-webpack-plugin@6.1.0" has unmet peer dependency "webpack@^4.4.0 || ^5.9.0".
warning "next-pwa > clean-webpack-plugin@3.0.0" has unmet peer dependency "webpack@*".
warning " > tailwindcss-font-inter@2.0.0" has incorrect peer dependency "tailwindcss@^1.9.0".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 23.34s.
jcgm@JoseloG-Macbook-Pro landing-site % yarn build          
yarn run v1.22.10
$ next build
Loaded env from /Users/jcgm/Dev/Kapital/landing-site/.env.local

> Build error occurred
Error: Cannot find module 'webpack'
Require stack:
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/generate-sw.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/index.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/next-pwa/index.js
- /Users/jcgm/Dev/Kapital/landing-site/next.config.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/next-server/server/config.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/build/index.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/cli/next-build.js
- /Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/generate-sw.js:18:17)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/index.js:10:20)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/generate-sw.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/workbox-webpack-plugin/build/index.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/next-pwa/index.js',
    '/Users/jcgm/Dev/Kapital/landing-site/next.config.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/next-server/server/config.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/build/index.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/cli/next-build.js',
    '/Users/jcgm/Dev/Kapital/landing-site/node_modules/next/dist/bin/next'
  ]
}
error Command failed with exit code 1.
`

### PR DESCRIPTION
**Pull request by Timer**
This pull request ensures the webpack hook is installed before an attempt is made to load the configuration.

This pull request is tested by the PnP tests, which should now be passing as a result of this change.

---

Fixes #21679

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 10.6s | 11.3s | ⚠️ +669ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.07 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.055 | 2.182 | ⚠️ +0.13 |
| / avg req/sec | 1216.76 | 1145.69 | ⚠️ -71.07 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.458 | 1.48 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1714.86 | 1689.31 | ⚠️ -25.55 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 13.1s | 13.5s | ⚠️ +363ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.07 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _error.js | 1.01 MB | 1.01 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.4 kB | 10.4 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.01 MB | 1.01 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.06 MB | 1.06 MB | ✓ |
| withRouter.js | 1.06 MB | 1.06 MB | ✓ |
| Overall change | 5.21 MB | 5.21 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.5s | ⚠️ +293ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.07 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.041 | 2.064 | ⚠️ +0.02 |
| / avg req/sec | 1224.96 | 1211.27 | ⚠️ -13.69 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.483 | 1.421 | -0.06 |
| /error-in-render avg req/sec | 1685.64 | 1759.12 | +73.48 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 597-74632e79..1d47.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 60.1 kB | 60.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-aedc815..1421.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-71ec2..1a96.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-33a09cb0..6745.js gzip | 536 B | 536 B | ✓ |
| hooks-4e968a..f870.js gzip | 902 B | 902 B | ✓ |
| index-5c6845..f75c.js gzip | 230 B | 230 B | ✓ |
| link-99f0c6c..b84a.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..bb56.js gzip | 306 B | 306 B | ✓ |
| withRouter-7..2133.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.57 kB | 8.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 326 B | 326 B | ✓ |
| Overall change | 326 B | 326 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| index.html gzip | 586 B | 586 B | ✓ |
| link.html gzip | 593 B | 593 B | ✓ |
| withRouter.html gzip | 581 B | 581 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "XAPia75-MmKFUYy6hxuig",
+        "buildId": "-8J1RqRJATefcVJNXCBmS",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_buildManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_ssgManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "XAPia75-MmKFUYy6hxuig",
+        "buildId": "-8J1RqRJATefcVJNXCBmS",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_buildManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_ssgManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "XAPia75-MmKFUYy6hxuig",
+        "buildId": "-8J1RqRJATefcVJNXCBmS",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_buildManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/XAPia75-MmKFUYy6hxuig/_ssgManifest.js"
+      src="/_next/static/-8J1RqRJATefcVJNXCBmS/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: b0884eccfadaaae68801c03bd78042c59d7ccb3f

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 12.9s | 13.3s | ⚠️ +388ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.08 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.516 | 2.464 | -0.05 |
| / avg req/sec | 993.57 | 1014.62 | +21.05 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.711 | 1.731 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1461.43 | 1443.99 | ⚠️ -17.44 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 15.6s | 15.6s | ⚠️ +48ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.08 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _error.js | 1.01 MB | 1.01 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.4 kB | 10.4 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.01 MB | 1.01 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.06 MB | 1.06 MB | ✓ |
| withRouter.js | 1.06 MB | 1.06 MB | ✓ |
| Overall change | 5.21 MB | 5.21 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| buildDuration | 15.5s | 15.7s | ⚠️ +166ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | ⚠️ +6.08 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.396 | 2.389 | -0.01 |
| / avg req/sec | 1043.44 | 1046.43 | +2.99 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.634 | 1.733 | ⚠️ +0.1 |
| /error-in-render avg req/sec | 1530.3 | 1442.65 | ⚠️ -87.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| 597-74632e79..1d47.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 60.1 kB | 60.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _app-aedc815..1421.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-71ec2..1a96.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-33a09cb0..6745.js gzip | 536 B | 536 B | ✓ |
| hooks-4e968a..f870.js gzip | 902 B | 902 B | ✓ |
| index-5c6845..f75c.js gzip | 230 B | 230 B | ✓ |
| link-99f0c6c..b84a.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..bb56.js gzip | 306 B | 306 B | ✓ |
| withRouter-7..2133.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.57 kB | 8.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 326 B | 326 B | ✓ |
| Overall change | 326 B | 326 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Timer/next.js hotfix/webpack-loading | Change |
| - | - | - | - |
| index.html gzip | 586 B | 586 B | ✓ |
| link.html gzip | 593 B | 593 B | ✓ |
| withRouter.html gzip | 581 B | 581 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "S4mRiW-xarCAoEsnW5hPF",
+        "buildId": "I6ialAHW4ps4W5EuVQ4jv",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_buildManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_ssgManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "S4mRiW-xarCAoEsnW5hPF",
+        "buildId": "I6ialAHW4ps4W5EuVQ4jv",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_buildManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_ssgManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "S4mRiW-xarCAoEsnW5hPF",
+        "buildId": "I6ialAHW4ps4W5EuVQ4jv",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_buildManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/S4mRiW-xarCAoEsnW5hPF/_ssgManifest.js"
+      src="/_next/static/I6ialAHW4ps4W5EuVQ4jv/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: 22378550991603ff955dc1b0c1e4b768147abca6

**Review by ijjk — APPROVED** [review]

File: packages/next/next-server/server/config.ts [hunk 1]
```diff
@@ -1,104 +1,18 @@
...
-  trailingSlash?: boolean
-
-  future: {
-    strictPostcssConfiguration: boolean
-    excludeDefaultMomentLocales: boolean
-    webpack5: boolean
-  }
-}
-
-const defaultConfig: NextConfig = {
```
Comment (balibebas) [inline packages/next/next-server/server/config.ts:42]: This removal just invalidated the docs, which shouldn't've been pointed to `canary` anyway: https://nextjs.org/docs/api-reference/next.config.js/introduction

Comment (Timer) [reply packages/next/next-server/server/config.ts:42]: Could you elaborate more? Those docs are still accurate!

**Pull request merged**